add fyi page about case sensitivity on macOS (#43)
expo/fyi: macos-ignorecase.md

> # case-insensitive filesystem on macOS

Reviewer: ```suggestion
# Default macOS filesystem is case-insensitive 
```

> By default filesystem on macOS is not case sensitive, for new (exlicitly created or copied/moved) files or directories the case of the filename is preserved when listing files, but it's ignored when you want to access specific file e.g. opening/deleting/moving. The situation might be a bit different when you refer to the file in code e.g. javascript import in react-native is case sensitive, because bundler reads exisiting files with correct case and all other operation happens inside the bundler in case sensitive way without filesystem interactions. If you create file `Example.tsx` importing it as `'./example'` won't work in react-native, but it would work in `node` becuse there is no bundler there and all imports use filesystem directly.

Reviewer: ```suggestion
The default macOS filesystem is case-insensitive. This means that for any new file or directory (explicitly created or copied/moved), the casing is preserved when listing files but it's ignored when you want to access the file. The situation might be a bit different when you refer to the file in your code, e.g. imports in React Native are case-sensitive. If you create file `Example.tsx` and import it as `'./example'`, it won't work in React Native. However, it would work in node.
```

> ### Syncing local case changes with state of the git repository 

Reviewer: ```suggestion
### Syncing local filesystem with the git repository state
```

> On macOS git option `core.ignorecase` is enabled by default, so e.g. 

Reviewer: ```suggestion
On macOS, the `core.ignorecase` git config option is enabled by default.
```

> If you change 

Reviewer: ```suggestion
So if you change:
```

 - filename `Example.tsx` -> `example.tsx`
 - all imports from `'./Example'` -> `'./example'` 

> then only import changes will be tracked by git, but file will still be called `Example.tsx`.

Reviewer: ```suggestion
then only the `import` changes will be tracked by git, but the filename change is ignored.
```

> As a result it will work for you locally, but it will fail for anyone that clones that repo including EAS Build that is using `git` to create shallow clone of the repo to upload project sources to the remote builders.

Reviewer: ```suggestion
As a result, it will work for you locally but it will fail for anyone that clones that repo (including EAS Build that is using `git` to create a shallow clone of the repo to upload project sources to the remote builders).
```

> To resolve issues caused by case inconsistency you need to either disable `core.ignorecase` by running `git config core.ignorecase false` or rename conflicitng files manually using `git mv`

Reviewer: ```suggestion
To resolve issues caused by the case inconsistencies you need to either disable `core.ignorecase` by running `git config core.ignorecase false` or rename problematic files manually using `git mv`.
```
